src/sagemaker/sklearn/processing.py

```diff
@@ -107,3 +108,64 @@ def __init__(
             tags=tags,
             network_config=network_config,
         )
+
+
+class SKLearnProcessorAlt(FrameworkProcessor):
+    """Handles Amazon SageMaker processing tasks for jobs using scikit-learn containers."""
+
+    estimator_cls = SKLearn
+
+    def __init__(
+        self,
+        framework_version,  # New arg
+        s3_prefix,  # New arg
+        role,
+        instance_count,
+        instance_type,
+        py_version="py3",  # New kwarg
+        image_uri=None,
+        volume_size_in_gb=30,
+        volume_kms_key=None,
+        output_kms_key=None,
+        max_runtime_in_seconds=None,
+        base_job_name=None,
+        sagemaker_session=None,
+        env=None,
+        tags=None,
+        network_config=None,
+    ):
+        """This processor executes a Python script in a scikit-learn execution environment.
+
+        This class has an 'Alt' suffix to denote it as an alternative to built-in
+        ``sagemaker.sklearn.processing.SKLearnProcessor``.
```

Comment:
What is the purpose of creating an alternative to the built-in `SKLearnProcessor`?

Reply:
If `FrameworkProcessor` anyway inherits from `ScriptProcessor`, wouldn't it make more sense to refactor the existing `SKLearnProcessor`? As far as I understand, this would not be a breaking change. Is that correct?

Reply:
Hi, thank you for the reviews. +1 to refactor `SKLearnProcessor` directly. Previously, I wasn't sure whether to change it, hence the Alt version.